modules/alookup/alookup.go

```diff
@@ -43,13 +44,47 @@ func (s *Lookup) DoLookup(name, nameServer string) (interface{}, zdns.Trace, zdn
 	return s.DoTargetedLookup(name, nameServer)
 }
 
+// Verify that A record is indeed IPv4 and AAAA is IPv6
+func verifyAddress(ansType string, ip string) bool {
+	isIpv4 := net.ParseIP(ip).To4() != nil
+	isIpv6 := net.ParseIP(ip).To16() != nil && strings.Contains(ip, ":")
```

Comment:
One more comment, if you want a minor optimization.  Since:
* You are always calling `To16()`
* `To16()` will guarantee, if non-nil, that `ip` is a valid IP

You don't need the call to `To4()`.  Logic like this would suffice and save a conversion call:

```go
var isIpv4, isIpv6 bool
if net.ParseIP(ip).To16() != nil {
    isIpv6 = strings.Contains(ip, ":")
    isIpv4 = !isIpv6
}
```

But even beyond that, I don't think the call to `To16()` is even needed; `ParseIP` will return `nil` if is not a valid textual representation of an IP address [1].  So I think this can get abbreviated down to:

```go
var isIpv4, isIpv6 bool
if net.ParseIP(ip) != nil {
    isIpv6 = strings.Contains(ip, ":")
    isIpv4 = !isIpv6
}
```

[1] https://pkg.go.dev/net#ParseIP

Reply:
Done. Also initialised isIpv4 and isIpv6 to false.